src/image.c

```diff
@@ -51,6 +55,63 @@ void image_rotate(struct image* ctx, size_t angle)
     }
 }
 
+// TODO: probably should be moved to config, to make use of expand_path()
+/**
+ * Get path to cached image thumbnail.
+ * @param image image
+ * @param path array in which the path should be stored
+ * @return true if successful
+ */
+static bool get_image_thumb_path(struct image* image, char* path)
+{
+    static char* cache_dir = NULL;
+    int r;
+    if (!cache_dir) {
+        cache_dir = getenv("XDG_CACHE_HOME");
+        if (cache_dir) {
+            cache_dir = strcat(cache_dir, "/swayimg");
+        } else {
+            cache_dir = getenv("HOME");
+            if (!cache_dir) {
+                return false;
+            }
+            cache_dir = strcat(cache_dir, "/.swayimg");
+        }
+    }
+    r = snprintf(path, PATH_MAX, "%s%s", cache_dir, image->source);
+    return r >= 0 && r < PATH_MAX;
+}
+
+// TODO: where should this function be?
+/**
+ * Makes directories like `mkdir -p`.
+ * @param path absolute path to the directory to be created
+ * @return true if successful
+ */
+static bool make_directories(char* path)
```

Comment:
`make_directory` in image entity. Something went wrong in program architecture =)

Reply:
I don't know where to put it...

Reply:
I would create `thumbnail.c` and move all these changes there (include pixman).